packages/editor/src/components/post-template/block-theme.js

```diff
@@ -24,25 +24,47 @@ const POPOVER_PROPS = {
 };
 
 export default function BlockThemeControl( { id } ) {
-	const { isTemplateHidden } = useSelect( ( select ) => {
-		const { getRenderingMode } = unlock( select( editorStore ) );
-		return {
-			isTemplateHidden: getRenderingMode() === 'post-only',
-		};
-	}, [] );
+	const { isTemplateHidden, getPostLinkProps, getEditorSettings, hasGoBack } =
+		useSelect( ( select ) => {
+			const { getRenderingMode, getEditorSettings: _getEditorSettings } =
+				unlock( select( editorStore ) );
+			const editorSettings = _getEditorSettings();
+			return {
+				isTemplateHidden: getRenderingMode() === 'post-only',
+				getPostLinkProps: editorSettings.getPostLinkProps,
+				getEditorSettings: _getEditorSettings,
+				hasGoBack: editorSettings.hasOwnProperty( 'goBack' ),
+			};
+		}, [] );
+
 	const { editedRecord: template, hasResolved } = useEntityRecord(
 		'postType',
 		'wp_template',
 		id
 	);
-	const { getEditorSettings } = useSelect( editorStore );
 	const { createSuccessNotice } = useDispatch( noticesStore );
 	const { setRenderingMode } = useDispatch( editorStore );
+	const editTemplate = getPostLinkProps
+		? getPostLinkProps( {
```

Comment:
I really don't like the name of the setting. I understand that it's needed because of the "href" but how important is it to have both the href and the onClick, `onSelectEntityRecord` would have been way better (or something like that).

Reply:
I agree. The naming was ok in the context of editing the synced patterns, but it hasn't extended well to this wider use case. As well as the setting naming, the `onClick` property is now also being used in places that don't involve a click.

I opened a [PR here](https://github.com/WordPress/gutenberg/pull/57745) with some suggested renaming options, initially `getEntityLoader`, but @talldan thought the use of `entity` [might be confused with the core-data apis](https://github.com/WordPress/gutenberg/pull/57745#issuecomment-1886263360).

That PR currently has the setting named as getPostNavigation, but the use of `post` is also not the best as we are using it to load templates and patterns. What about `getEntityNavigation` with properties of `goTo` and `linkTo` ?

Reply:
What I don't like is the "get" prefix, I prefer a callback directly "on" something, so my question is can't we get rid of the href and just leave the onClick?

Reply:
I actually think `href` is more important than `onClick` in terms of universal routing. In a future context, we might be able to just use regular `<a>` tags and `href` attributes for the post editor or shared packages (`core-commands`). In the site editor, we can attach an event listener globally that listens to all the bubbled events with the `<a>` links. Then, we can intercept the click and use `history.push()` under the hood to achieve seamless client-side routing.

`href` provides more context in this case, and it's accessible from the browsers' viewpoint as well (the bottom-left address when you hover over it and open-in-new-tab functionality, for instance.)

Reply:
> I actually think href is more important than onClick in terms of universal routing. 

I agree when it comes to links. The problem is that this is an API of the "block-editor" package and these actions are not always links.

Reply:
The href was initially working in the post editor when we were updating the url params, but that caused other issues, so href has been broken in the post editor since we stopped updating the url. So, I have removed the href and renamed the setting to your suggestion @youknowriad of `onSelectEntityRecord`. There is a [PR for this](https://github.com/WordPress/gutenberg/pull/58416) here. 

I haven't extensively tested this yet, I will wait and see if you are happy with this new naming approach first.